src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -213,25 +214,33 @@ private[spark] case class CosmosDBConnection(config: Config) extends CosmosDBLog
   }
 
   def getCollectionThroughput: Int = {
-    var offers = documentClient.queryOffers(s"SELECT * FROM c where c.offerResourceId = '${getCollection.getResourceId}'", null).getQueryIterable.toList
-    if (offers.isEmpty) {
-      offers = documentClient.queryOffers(s"SELECT * FROM c where c.offerResourceId = '${getDatabase.getResourceId}'", null).getQueryIterable.toList
-      // database throughput
-      if (offers.isEmpty) {
-          throw new IllegalStateException("Cannot find the collection corresponding offer.")
-      }
-    }
+    collectionThroughput match {
+      case Some(value) => value
```

Comment:
If the customer changes the offer settings externally (i.e. Azure Portal) this cached value will not reflect the new value therefore requiring an app restart. Alternatively we could cache the current time and throughput value and only refresh every 10 minutes or something like that.

Reply:
Understood - but the manual configuration can also only be updated on restart. For V3 I will use a pull-based model to get meta data - not worth making it more complicated here